packages/block-editor/src/store/selectors.js

```diff
@@ -1971,7 +1972,7 @@ export const getInserterItems = createSelector(
 			};
 		};
 
-		const blockTypeInserterItems = getBlockTypes()
+		const blockTypeInserterItems = getBlockTransformItems()
 			.filter( ( blockType ) =>
 				canIncludeBlockTypeInInserter( state, blockType, rootClientId )
 			)
```

Comment:
The Page List transform gets filtered out here 🤔

Reply:
Resolved by https://github.com/WordPress/gutenberg/pull/46426#issuecomment-1348138164